src/app/data/loaders/cdfLoader.ts

```diff
@@ -0,0 +1,119 @@
+import { v4 } from '@lukeed/uuid';
+import type { FileCollection } from 'filelist-utils';
+import type { Value } from 'cheminfo-types';
+import { NetCDFReader } from 'netcdfjs';
+import { xMaxValue } from 'ml-spectra-processing';
+
+import {
+  MeasurementKind,
+  Loader,
+  Measurements,
+  getEmptyMeasurements,
+} from '../DataState';
+
+export const cdfLoader: Loader = async function cdfLoader(
+  fileCollection: FileCollection,
+) {
+  const newMeasurements: Measurements = getEmptyMeasurements();
+  for (const file of fileCollection) {
+    if (file.name.match(/(?:\.cdf)$/i)) {
+      const reader = new NetCDFReader(await file.arrayBuffer(), { meta: true });
+      let kind: MeasurementKind | undefined;
+      if (
+        reader.dataVariableExists('mass_values') &&
+        reader.dataVariableExists('time_values')
+      ) {
+        kind = 'gclcms';
+      }
+
+      const time = reader.getDataVariable('scan_acquisition_time');
+      const tic = reader.getDataVariable('total_intensity');
+      const pointCount = reader.getDataVariable('point_count');
+      const massValues = reader.getDataVariable('mass_values');
+      const intensityValues = reader.getDataVariable('intensity_values');
+      addMeta(reader, reader.globalAttributes);
+
+      let index = 0;
+      for (let i = 0; i < pointCount.length; i++) {
+        const size = pointCount[i];
+        const mass = new Array(size);
+        const intensity = new Array(size);
+        for (let j = 0; j < size; j++) {
+          mass[j] = massValues[index];
+          intensity[j] = intensityValues[index++];
+        }
+        if (kind) {
+          newMeasurements[kind].entries.push({
+            id: v4(),
+            meta: reader.header.meta,
+            filename: file.name,
+            path: file.relativePath || '',
+            info: reader.getAttribute('experiment_title'),
+            title: reader.getAttribute('experiment_title'),
+            data: normalizeChromatogram({
+              time: time[i],
```

Comment:
The data should be an array, one entry per time.

In the info:
https://github.com/zakodium-oss/analysis-ui-components/blob/43adbe8737760eab6db24acca0ed53a588716a4d/src/app/data/MeasurementBase.ts#L20 we can have tic and bpc.

Reply:
So, it's ok?

Reply:
You store indeed in the right place but for one GC/MS there should be only one measurement entries and each 'time' is an element in the data array.

https://github.com/zakodium-oss/analysis-ui-components/blob/43adbe8737760eab6db24acca0ed53a588716a4d/src/app/data/MeasurementBase.ts#L18-L22